fog/ledger/server/tests/router_connection.rs

```diff
@@ -88,23 +88,23 @@ fn fog_ledger_merkle_proofs_test(logger: Logger) {
             &recipients,
             &[],
             &mut rng,
-            &mut watcher,
+            &watcher,
```

Comment:
The watcher db `add` methods aren't mut...? https://github.com/mobilecoinfoundation/mobilecoin/blob/f957c956e67ce28a6cc5f571d28e6e364e3f7f2e/watcher/src/watcher_db.rs#L269

It kind of feels like they should be, but they're interacting with a state outside of rust where the one mutable instance may not make as much sense.

Reply:
I don't think the lack of a`mut` necessarily implies no state is being manipulated (e.g. anything that locks a mutex can then manipulate state and the mutex doesn't need a `mut` reference to be locked).